library/analytics/src/main/java/org/cru/godtools/analytics/model/ExitLinkActionEvent.kt

```diff
@@ -1,12 +1,14 @@
 package org.cru.godtools.analytics.model
 
 import android.net.Uri
+import android.os.Bundle
 
-private const val ACTION_EXIT_LINK = "Exit Link Engaged"
-private const val ADOBE_EXIT_LINK = "cru.mobileexitlink"
+private const val ACTION_EXIT_LINK = "exit_link_engaged"
+private const val FIREBASE_PARAM_EXIT_LINK = "cru_mobileexitlink"
 
 class ExitLinkActionEvent(link: Uri) : AnalyticsActionEvent(action = ACTION_EXIT_LINK) {
-    override fun isForSystem(system: AnalyticsSystem) = system == AnalyticsSystem.ADOBE
-    override val adobeAttributes = mapOf<String, Any>(ADOBE_EXIT_LINK to link.toString())
-    override val firebaseEventName = "exit_link_engaged"
+    override fun isForSystem(system: AnalyticsSystem) = system == AnalyticsSystem.FIREBASE
+    override val firebaseParams = Bundle().apply {
+        putString(FIREBASE_PARAM_EXIT_LINK, link.toString())
+    }
```

Comment:
```suggestion
    override val firebaseParams get() = Bundle().apply {
        putString(FIREBASE_PARAM_EXIT_LINK, link.toString())
    }
```

something subtle you should do here is define the property with a getter, you will also have to make the `link: Uri` constructor param a `private val`.

Using the `get()` syntax will regenerate a fresh bundle every time it is called. otherwise it will generate the Bundle once when the object is initially created, and always return the same `Bundle`. Because `Bundles` are mutable, any changes made to the returned `Bundle` will persist for subsequent calls to `firebaseParams`.

We currently don't modify the returned Bundle, but I can't say that will always be the case.

Reply:
it was one trade-off I made when switching from `Map` to `Bundle`

Reply:
Would creating the ```private val``` just be doing something like ```private val link = link```?

Reply:
```kotlin
class ExitLinkActionEvent(private val link: Uri) : AnalyticsActionEvent(action = ACTION_EXIT_LINK) {
```